src/lib/download/downloadFiles.ts

```diff
@@ -35,6 +35,16 @@ async function downloadFileWithFetch(
   }
 }
 
+export function filterToParams(filter: Record<string, any>): string {
```

Comment:
If we're making a new version of `filterToParams`, we could also consider using [the URL API for query parameters](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams), which might be a bit nicer to work with than strings?